src/App.tsx

```diff
@@ -14,11 +14,84 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
-
-export default App;
+type State = {
+  selectedGoods: string[];
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: ['Jam'],
+  };
+
+  onSelect = (good: string) => {
+    this.setState(state => ({
+      selectedGoods: [...state.selectedGoods, good],
+    }));
+  };
+
+  onDelete = (good: string) => {
+    this.setState(state => ({
+      selectedGoods: state.selectedGoods.filter(item => item !== good),
+    }));
+  };
+
+  deleteAll = () => {
+    this.setState({
+      selectedGoods: [],
+    });
+  };
+
+  listToShow = () => {
+    if (this.state.selectedGoods.length === 1) {
+      return `${this.state.selectedGoods.join(', ')} is selected`;
+    }
+
+    return `${this.state.selectedGoods.join(', ')} are selected`;
```

Comment:
I believe you can do it better, and improve the` listToShow` function to print `good1, good2 and lastgood` , not only a list separated with coma

Reply:
Thanks for the feedback, Valerii! At the expense of styles, I wanted to practice, because I worked with css for a long time) In the future I need to try some framework). At the expense of the comment, your opinion is good. Maybe on the weekend I'll come up with something))